lib/syntax.ml

```diff
@@ -278,6 +287,12 @@ and assign_types expr =
         | Ret ret, _ ->
           let nullability = common_nullability types in
           { t = ret; nullability; }, types (* ignoring arguments FIXME *)
+        | Comparison, _  ->
+          let args, ret = convert_args (Typ (depends Bool)) [Var 0; Var 0] in
+          let nullable = common_nullability args in
+          if param_strict && List.exists is_param params then 
+            make_strict ret, List.map make_strict args 
```

Comment:
this can be done before convert_args and I suspect it will result in better propagation

Reply:
For the returning type I changed it, but kept it for args